File: SER321-master/Assigns/Assign2/src/java/MovieLibrary.java
import java.util.List;
import java.util.ArrayList;

public class MovieLibrary
{
  private List<MovieDescription> theLibrary = new ArrayList<MovieDescription>();

  public boolean add(MovieDescription aClip)
  {
    return theLibrary.add(aClip);
  }

  public boolean remove(String aTitle)
  {
    return theLibrary.remove(get(aTitle));
  }

  public MovieDescription get(String aTitle)
  {
    for (int i = 0; i < theLibrary.size(); i++)
    {
      if (theLibrary.get(i).getTitle() == aTitle)
      {
        return theLibrary.get(i);
      }
    }

    return null;
  }

  public String[] getTitles()
  {
    String[] titles = new String[theLibrary.size()];

    for (int i = 0; i < theLibrary.size(); i++)
    {
      titles[i] = theLibrary.get(i).getTitle();
    }

    return titles;
  }
}

File: SER321-master/Assigns/Assign2/src/cpp/MovieLibrary.cpp
#include "MovieLibrary.hpp"
#include <string>
#include <vector>
using namespace std;

class MovieDescription;

bool MovieLibrary::add(MovieDescription aClip)
{
  theLibrary.push_back(aClip);

  if (theLibrary.back().getTitle() == aClip.getTitle())
  {
    return true;
  }

  return false;
}

bool MovieLibrary::remove(string aTitle)
{
  for (int i = 0; i < theLibrary.size(); i++)
  {
    if (theLibrary.at(i).getTitle() == aTitle)
    {
      theLibrary.erase(theLibrary.begin() + i);
      return true;
    }
  }

  return false;
}

MovieDescription MovieLibrary::get(string aTitle)
{
  for (int i = 0; i < theLibrary.size(); i++)
  {
    if (aTitle == theLibrary.at(i).getTitle())
    {
      return theLibrary.at(i);
    }
  }
}

string* MovieLibrary::getTitles()
{
  string* titles = new string[theLibrary.size()];

  for (int i = 0; i < theLibrary.size(); i++)
  {
    titles[i] = theLibrary.at(i).getTitle();
  }

  return titles;
}
